netcdf 4.9.0 test failures with gcc 11.3

Need to check if this manifests with other gcc compiler versions.

netcdf is also now at version 4.9.2 and MSYS2 are applying some additional build options.
https://github.com/msys2/MINGW-packages/commit/8d0bb1753bb2b303ce02a299f06c0735ffb5a5ce


```
Making check in unit_test
make[1]: Entering directory '/z/extlib/_bprogressive__.src/netcdf-c-4.9.0/unit_test'
make  tst_nclist.exe test_ncuri.exe test_pathcvt.exe tst_exhash.exe tst_xcache.exe tst_nc4internal.exe
make[2]: Entering directory '/z/extlib/_bprogressive__.src/netcdf-c-4.9.0/unit_test'
depbase=`echo tst_nclist.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
x86_64-w64-mingw32-gcc -DHAVE_CONFIG_H -I. -I..  -I../include -I../include   -I/z/extlib/_bprogressive__/include  -fno-s
trict-aliasing -MT tst_nclist.o -MD -MP -MF $depbase.Tpo -c -o tst_nclist.o tst_nclist.c &&\
mv -f $depbase.Tpo $depbase.Po
/bin/sh ../libtool  --tag=CC   --mode=link x86_64-w64-mingw32-gcc  -fno-strict-aliasing   -L/z/extlib/_bprogressive__/li
b -o tst_nclist.exe tst_nclist.o ../liblib/libnetcdf.la -ljpeg -lmfhdf -lhdf -ljpeg -lhdf5_hl -lhdf5 -lm -lzlib -ldl -ls
z -lbz2 -lxml2
libtool: link: x86_64-w64-mingw32-gcc -fno-strict-aliasing -o .libs/tst_nclist.exe tst_nclist.o  -L/z/extlib/_bprogressi
ve__/lib ../liblib/.libs/libnetcdf.dll.a -lmfhdf -lhdf -ljpeg -lhdf5_hl -lhdf5 -lzlib -ldl /z/extlib/_bprogressive__/lib
/libsz.dll.a -lbz2 -lxml2 -L/z/extlib/_bprogressive__/lib
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0x146): undefined reference to `count_NCList'
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0x1c1): undefined reference to `free_NCList'
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0x1d0): undefined reference to `find_in_NCList_by_name'
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0x256): undefined reference to `iterate_NCList'
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0x353): undefined reference to `new_NC'
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0x3d8): undefined reference to `find_in_NCList'
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0x45b): undefined reference to `add_to_NCList'
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0x465): undefined reference to `find_in_NCList'
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0x4eb): undefined reference to `find_in_NCList_by_name'
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0x573): undefined reference to `iterate_NCList'
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0x5fb): undefined reference to `find_in_NCList'
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0x689): undefined reference to `find_in_NCList_by_name'
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0x719): undefined reference to `iterate_NCList'
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0x799): undefined reference to `count_NCList'
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0x815): undefined reference to `free_NCList'
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0x82a): undefined reference to `del_from_NCList'
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0x836): undefined reference to `free_NC'
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0x840): undefined reference to `find_in_NCList'
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0x93e): undefined reference to `new_NC'
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0x9c5): undefined reference to `add_to_NCList'
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0x9d2): undefined reference to `find_in_NCList'
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0xa69): undefined reference to `move_in_NCList'
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0xae9): undefined reference to `find_in_NCList'
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0xb75): undefined reference to `del_from_NCList'
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0xb81): undefined reference to `free_NC'
z:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: tst_nclist.o:tst_nclist.c
:(.text+0xb8b): undefined reference to `find_in_NCList'
collect2.exe: error: ld returned 1 exit status
make[2]: *** [Makefile:768: tst_nclist.exe] Error 1
make[2]: Leaving directory '/z/extlib/_bprogressive__.src/netcdf-c-4.9.0/unit_test'
make[1]: *** [Makefile:1120: check-am] Error 2
make[1]: Leaving directory '/z/extlib/_bprogressive__.src/netcdf-c-4.9.0/unit_test'
make: *** [Makefile:764: check-recursive] Error 1

```

--- a/build.sh
+++ b/build.sh
@@ -1140,16 +1140,18 @@ cd $WRKDIR/$PACK
 
 ###this is a hack
 sed -i s/-ldf/-lhdf/ configure
-sed -i 's/libnetcdf_la_LDFLAGS = /libnetcdf_la_LDFLAGS = -no-undefined /' liblib/Makefile.in
+#sed -i 's/libnetcdf_la_LDFLAGS = /libnetcdf_la_LDFLAGS = -no-undefined /' liblib/Makefile.in
 
 save_configure_help
-CPPFLAGS=-I$OUTINC LDFLAGS=-L$OUTLIB xxrun ./configure $HOSTBUILD --prefix=$OUT --enable-static=no --enable-shared=yes \
+CPPFLAGS=-I$OUTINC LDFLAGS="-L$OUTLIB -Wl,--export-all-symbols" \
+    xxrun ./configure $HOSTBUILD --prefix=$OUT --enable-static=no --enable-shared=yes \
                         --enable-hdf4 --disable-dap  --disable-dynamic-loading \
                        --disable-utilities --disable-plugins \
-                       --disable-nczarr-filters --disable-nczarr
+                       --disable-nczarr-filters --disable-nczarr \
+                       --disable-byterange
 patch_libtool
-xxrun make
-xxrun make check
+xxrun make 
+xxrun make check 
 xxrun make install
 ;;
 

--- a/patches/netcdf-c-4.9.2/0001-no-debug-libraries.patch
+++ b/patches/netcdf-c-4.9.2/0001-no-debug-libraries.patch
@@ -1,0 +1,48 @@
+diff -urN netcdf-c-4.9.0/cmake/modules/FindZip.cmake.orig netcdf-c-4.9.0/cmake/modules/FindZip.cmake
+--- netcdf-c-4.9.0/cmake/modules/FindZip.cmake.orig	2022-06-23 17:20:15.443959500 +0200
++++ netcdf-c-4.9.0/cmake/modules/FindZip.cmake	2022-06-23 17:20:24.156687000 +0200
+@@ -39,7 +39,7 @@
+                PATHS ${Zip_LIBRARY_DIRS} NO_DEFAULT_PATH)
+ 
+   SET(Zip_LIBRARIES )
+-  IF(Zip_DEBUG_LIBRARY AND Zip_RELEASE_LIBRARY)
++  IF(Zip_DEBUG_LIBRARY AND Zip_RELEASE_LIBRARY AND NOT (Zip_DEBUG_LIBRARY STREQUAL Zip_RELEASE_LIBRARY))
+     SET(Zip_LIBRARIES debug ${Zip_DEBUG_LIBRARY} optimized ${Zip_RELEASE_LIBRARY})
+   ELSEIF(Zip_DEBUG_LIBRARY)
+     SET(Zip_LIBRARIES ${Zip_DEBUG_LIBRARY})
+diff -urN netcdf-c-4.9.0/cmake/modules/FindBlosc.cmake.orig netcdf-c-4.9.0/cmake/modules/FindBlosc.cmake
+--- netcdf-c-4.9.0/cmake/modules/FindBlosc.cmake.orig	2022-06-23 17:59:01.868937600 +0200
++++ netcdf-c-4.9.0/cmake/modules/FindBlosc.cmake	2022-06-23 18:07:01.844880800 +0200
+@@ -39,7 +39,7 @@
+                PATHS ${Blosc_LIBRARY_DIRS} NO_DEFAULT_PATH)
+ 
+   SET(Blosc_LIBRARIES )
+-  IF(Blosc_DEBUG_LIBRARY AND Blosc_RELEASE_LIBRARY)
++  IF(Blosc_DEBUG_LIBRARY AND Blosc_RELEASE_LIBRARY AND NOT (Blosc_DEBUG_LIBRARY STREQUAL Blosc_RELEASE_LIBRARY))
+     SET(Blosc_LIBRARIES debug ${Blosc_DEBUG_LIBRARY} optimized ${Blosc_RELEASE_LIBRARY})
+   ELSEIF(Blosc_DEBUG_LIBRARY)
+     SET(Blosc_LIBRARIES ${Blosc_DEBUG_LIBRARY})
+diff -urN netcdf-c-4.9.0/cmake/modules/FindZstd.cmake.orig netcdf-c-4.9.0/cmake/modules/FindZstd.cmake
+--- netcdf-c-4.9.0/cmake/modules/FindZstd.cmake.orig	2022-06-10 23:04:15.000000000 +0200
++++ netcdf-c-4.9.0/cmake/modules/FindZstd.cmake	2022-06-23 18:10:24.665696300 +0200
+@@ -39,7 +39,7 @@
+                PATHS ${Zstd_LIBRARY_DIRS} NO_DEFAULT_PATH)
+ 
+   SET(Zstd_LIBRARIES )
+-  IF(Zstd_DEBUG_LIBRARY AND Zstd_RELEASE_LIBRARY)
++  IF(Zstd_DEBUG_LIBRARY AND Zstd_RELEASE_LIBRARY AND NOT (Zstd_DEBUG_LIBRARY STREQUAL Zstd_RELEASE_LIBRARY))
+     SET(Zstd_LIBRARIES debug ${Zstd_DEBUG_LIBRARY} optimized ${Zstd_RELEASE_LIBRARY})
+   ELSEIF(Zstd_DEBUG_LIBRARY)
+     SET(Zstd_LIBRARIES ${Zstd_DEBUG_LIBRARY})
+diff -urN netcdf-c-4.9.0/cmake/modules/FindBz2.cmake.orig netcdf-c-4.9.0/cmake/modules/FindBz2.cmake
+--- netcdf-c-4.9.0/cmake/modules/FindBz2.cmake.orig	2022-06-10 23:04:15.000000000 +0200
++++ netcdf-c-4.9.0/cmake/modules/FindBz2.cmake	2022-06-23 18:13:29.254600600 +0200
+@@ -39,7 +39,7 @@
+                PATHS ${Bz2_LIBRARY_DIRS} NO_DEFAULT_PATH)
+ 
+   SET(Bz2_LIBRARIES )
+-  IF(Bz2_DEBUG_LIBRARY AND Bz2_RELEASE_LIBRARY)
++  IF(Bz2_DEBUG_LIBRARY AND Bz2_RELEASE_LIBRARY AND NOT (Bz2_DEBUG_LIBRARY STREQUAL Bz2_RELEASE_LIBRARY))
+     SET(Bz2_LIBRARIES debug ${Bz2_DEBUG_LIBRARY} optimized ${Bz2_RELEASE_LIBRARY})
+   ELSEIF(Bz2_DEBUG_LIBRARY)
+     SET(Bz2_LIBRARIES ${Bz2_DEBUG_LIBRARY})

